models/product/src/product-data/builders.spec.ts

```diff
@@ -0,0 +1,299 @@
+      name: expect.any(String),
+      nameAllLocales: expect.arrayContaining([
+        expect.objectContaining({
+          locale: 'en',
+          value: expect.any(String),
+          __typename: 'LocalizedString',
+        }),
+      ]),
+      description: expect.any(String),
+      descriptionAllLocales: expect.arrayContaining([
+        expect.objectContaining({
+          locale: 'en',
+          value: expect.any(String),
+          __typename: 'LocalizedString',
+        }),
+      ]),
+      slug: expect.any(String),
+      slugAllLocales: expect.arrayContaining([
+        expect.objectContaining({
+          locale: 'en',
+          value: expect.any(String),
+          __typename: 'LocalizedString',
+        }),
+      ]),
+      categoryOrderHint: expect.any(String),
+      categoryOrderHints: expect.arrayContaining([
+        expect.objectContaining({
+          __typename: 'CategoryOrderHint',
+          categoryId: expect.any(String),
+          orderHint: expect.any(String),
+        }),
+      ]),
+      categoriesRef: expect.arrayContaining([
+        {
+          typeId: 'category',
+          id: expect.any(String),
+          __typename: 'Reference',
+        },
+      ]),
+      categories: expect.arrayContaining([
+        expect.objectContaining({
+          id: expect.any(String),
+          key: expect.any(String),
+          __typename: 'Category',
+        }),
+      ]),
+      searchKeyword: expect.arrayContaining([
+        expect.objectContaining({
+          text: expect.any(String),
+        }),
+      ]),
+      searchKeywords: expect.arrayContaining([
+        expect.objectContaining({
+          searchKeywords: expect.arrayContaining([
+            expect.objectContaining({
+              text: expect.any(String),
+            }),
+          ]),
+        }),
+      ]),
+      metaTitle: expect.any(String),
+      metaTitleAllLocales: expect.arrayContaining([
+        expect.objectContaining({
+          locale: 'en',
+          value: expect.any(String),
+          __typename: 'LocalizedString',
+        }),
+      ]),
+      metaKeywords: expect.any(String),
+      metaKeywordsAllLocales: expect.arrayContaining([
+        expect.objectContaining({
+          locale: 'en',
+          value: expect.any(String),
+          __typename: 'LocalizedString',
+        }),
+      ]),
+      metaDescription: expect.any(String),
+      metaDescriptionAllLocales: expect.arrayContaining([
+        expect.objectContaining({
+          locale: 'en',
+          value: expect.any(String),
+          __typename: 'LocalizedString',
+        }),
+      ]),
+      masterVariant: expect.objectContaining({ id: expect.any(Number) }),
+      variant: expect.objectContaining({ id: expect.any(Number) }),
+      variants: expect.arrayContaining([
+        expect.objectContaining({ id: expect.any(Number) }),
+      ]),
+      allVariants: expect.arrayContaining([
+        expect.objectContaining({ id: expect.any(Number) }),
+      ]),
```

Comment:
When validating GraphQL models, we usually rely on the `__typename` attribute. That should be enough from the model point of view.
We are using that attribute in other properties of this model so I was wondering why we don't use it for all of them.